src/consumer.cpp

```diff
@@ -200,6 +200,24 @@ Consumer::get_offsets_position(const TopicPartitionList& topic_partitions) const
     return convert(topic_list_handle);
 }
 
+void Consumer::store_offsets() const
```

Comment:
I'm not sure about this overload. When would you use it? Normally you either want to "commit" (in memory in this case) a message or a topic/partition/offset list. Is this meant to be used by just consuming messages, not doing any commit/offset store and then just call this and the offsets get magically committed in memory? If so, I'd probably change this to something like `store_seen_offsets` or `store_consumed_offsets` so it's more in line with what it does. I feel like `store_offsets` without any parameter is not very descriptive.

Also, minor comment: please put braces on the same line as the method definition.

Reply:
This was more as a helper function similar to what the `commit()` does, which uses the assigned partitions under the hood. Since `commit` has 3 overloads I though, for completeness sake, to also have 3 overloads there. I will change it to `store_consumed_offsets`.  
None of the `store_offset` overloads work by themselves, you need to have the `enable.auto.commit=true` so that it actually commits them to the broker. I can also add that in the comment. 

>Also, minor comment: please put braces on the same line as the method definition.

yes, i keep making this mistake all the time :)

Reply:
> I'm not sure about this overload. When would you use it?

Say you consume a batch of messages. Then simply call `store_consumed_offsets()` periodically or after the batch is done.

Reply:
Fair enough. Looks like [this is failing](https://travis-ci.org/mfontanini/cppkafka/builds/489666899?utm_source=github_status&utm_medium=notification) on rdkafka 0.9.4, so we may need to use some preprocessor guard before using `rd_kafka_offsets_store`. Looks like it was introduced [in v0.11.0[(https://github.com/edenhill/librdkafka/commit/715c49bfbf9f3fe41934989cf92408966949954e)

Reply:
Indeed I hadn't noticed the rdkafka version was different for clang and gcc.